webblog/frontend/src/reducers/auth.js

```diff
@@ -0,0 +1,57 @@
+import {
+    AUTH_ERROR,
+    LOGIN_FAIL,
+    LOGIN_SUCCESS,
+    LOGOUT_SUCCESS,
+    REGISTER_FAIL,
+    REGISTER_SUCCESS,
+    USER_LOADED,
+    USER_LOADING
+} from "../actions/types";
+
+const initialState = {
+    token: localStorage.getItem("token"),
+    isAuthenticated: null,
+    isLoading: false,
+    user: null
+};
+
+export default function (state = initialState, action) {
+    switch (action.type) {
+        case USER_LOADING:
+            return {
+                ...state,
```

Comment:
What is the reason of using ...state syntax ?

Reply:
this reducer set rest of the states to redux state store.